### ISSUE TITLE
[react-hooks] 1.12.0 minor broke import in Next.js

### ISSUE THREAD

**Issue by qw-in** [Shopify/quilt#1741]
## Overview

The `1.11` to `1.12` minor update changed the way `react-hooks` is authored and broke the import for a client's next.js app. I have downgraded for now. It appears to be due to the `.mjs` file.

This is likely just growing pains with support for `.mjs` and bundlers catching up (webpack is used internally by next) however I think either way this change could have been a major.

Likely no action required, but thought I would open in case anyone else runs into this. Some searchable stuff below:

```
Can't import the named export '<hook>' from non EcmaScript module (only default export is available)
```

## Consuming repo

> What repo were you working in when this issue occurred?
Private repo, can provide reproduction repo at request

### Area
Not sure :thinking: 

### Scope
react-hooks

**Comment by GoodForOneFare** [comment]
cc @BPScott @ismail-syed - I think you're looking into this?

**Comment by ismail-syed** [comment]
Yes, I've been seeing related reports. I'm going to spend some time next week to look into this deeper.

**Comment by ismail-syed** [comment]
Agreed, growing pains of `.mjs` support. There's no real action here to take. In retrospect, this should have been a breaking change. For adding support in nextjs configs, see https://github.com/Shopify/quilt/issues/1722#issuecomment-756963544.
